dash/backend/src/modules/falco/dao/falco.dao.ts

```diff
@@ -113,12 +120,58 @@ export class FalcoDao {
         }
     }
 
+    async getFalcoLogByEventId(
+       eventId: number
+    ): Promise<FalcoDto> {
+        const knex = await this.databaseService.getConnection();
+
+       const result= knex.select(
+            ['p.id as id', 'p.calendar_date as calendarDate', 'p.namespace as namespace', 'p.container as container',
+                'p.image as image', 'p.message as message', 'p.anomaly_signature as anomalySignature', 'p.raw as raw'])
+            .from('project_falco_logs AS p')
+            .where('p.id', eventId)
+            .then(result => result[0]);
+
+         return result;
+    }
+
+    async getCountOfFalcoLogsBySignature(
+        clusterId: number, signature: string
+    ): Promise<FalcoCountDto[]> {
+
+        const currentDate = set(new Date(), {hours: 0, minutes: 0, seconds: 0, milliseconds: 0});
+        const endDate = format(currentDate, 'yyyy-MM-dd');
+        const startDate = format(sub(currentDate, {days: 28}), 'yyyy-MM-dd');
+
+        const knex = await this.databaseService.getConnection();
+
+        // get all signature logs within 28 day from current date
+        const signatureLogs= knex.select()
+            .from('project_falco_logs')
+            .where('anomaly_signature', signature)
+            .andWhere('cluster_id', clusterId)
+            .andWhere('calendar_date', '>=', startDate)
+            .andWhere('calendar_date', '<=', endDate);
+
+
+        const signatureCountByDate = await knex
+            .select( [knex.raw( 'calendar_date as date, count (calendar_date)')])
```

Comment:
One thing to consider is if it occurs on Monday and Wednesday, we may still want the API to return 0 for Tuesday. So, we may want to populate days with 0 results, too, in the graph (either on frontend or backend).

Reply:
I added this with JS instead of using query.